### DISCUSSION TITLE
Run next build in development mode

### DISCUSSION THREAD

**Discussion by LucaWishraiser** [vercel/next.js#45611] [Help]
Hi,

I would like to start my NextJS application on Vercel ( preview mode ) with libraries (like React) running in dev mode and enable source maps. Currently when running in preview mode all is set with production mode and that leads to unable to debug errors.

I saw this old post from 2019 but I don't know if there are any updates on this..

https://github.com/vercel/next.js/issues/6062

**Comment by icyJoseph** [comment]
> unable to debug errors

What error in particular? 

I've seen people claim that `NODE_ENV` can be modified during build time and start time, I tested it and it didn't work for me, but even if it did work locally, one issue is that in development mode a server is used to stream changes and bundles down to the client, but when you host a Next.js app on Vercel, it uses a serverless mode. 

If all of your pages were GSSP and used API routes, then on every request a lambda is created (if not available already), and it handles the GSSP request, or API route request. If the resources are static the HTML/JS/CSS files are served directly. 

I think it might be easier to just try to debug the problem in production code. If its one of those hydration errors issues, it might be tricky, but sometimes it is possible to find the actual DOM node that's causing the issue.
